bears/python/PEP8JupyterNotebookBear.py

```diff
@@ -0,0 +1,128 @@
+import autopep8
+from copy import copy
+import nbformat
+
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from coalib.bears.LocalBear import LocalBear
+from coalib.bears.requirements.PipRequirement import PipRequirement
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.settings.Setting import typed_list
+
+
+class NotebookHandler:
+    """Utility class for Jupyter Notebook handling."""
+
+    def __init__(self, filename, file):
+        """
+        Constructs a notebook handler.
+
+        :param filename: The filename of the file
+        :param file:     The file contents as string array
+        """
+        self.filename = filename
+        self.file = file
+        self._file_dct = self.read_from_list(file)
+
+    @staticmethod
+    def read_from_list(file):
+        """
+        Reads notebook in string array form into string.
+
+        :param file:     The notebook file contents as string array
+        :return:         The notebook file as string
+        """
+        return nbformat.reads(''.join(file), nbformat.NO_CONVERT)
+
+    def write_to_list(self, file_dct):
+        """
+        Writes notebook in dict form to string array.
+
+        :param file_dct: The notebook in dictionary form.
+        :return:         The notebook in string array form.
+        """
+        file_lst = (nbformat.writes(file_dct, nbformat.NO_CONVERT)
+                            .splitlines(True))
+        # nbformat.writes does not add a final \n, we do this manually
+        # necessary for the diff because original file has it
+        if self.file[-1].endswith('\n') and not file_lst[-1].endswith('\n'):
```

Comment:
youu could add a testcase without a trailing newline at all to cover this branch